gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-mqtt5/src/main/java/io/gravitee/plugin/endpoint/mqtt5/Mqtt5EndpointConnector.java

```diff
@@ -249,7 +273,48 @@ private String getResponseTopic(final ExecutionContext ctx) {
         return responseTopic;
     }
 
+    private String overrideConfigFromMessage(final String sharedConfiguration, final Message message, final String attributeKey) {
+        String returnedConfiguration = sharedConfiguration;
+        String attributeConfiguration = message.attribute(attributeKey);
+        if (attributeConfiguration != null && !attributeConfiguration.isEmpty()) {
+            returnedConfiguration = attributeConfiguration;
+        }
+        if (returnedConfiguration != null) {
+            message.attribute(attributeKey, returnedConfiguration);
+        }
+        return returnedConfiguration;
+    }
+
+    private Flowable<Message> interruptMessagesWith(final ExecutionContext ctx, final Throwable throwable) {
+        if (
+            throwable instanceof ConnectionFailedException ||
+            throwable instanceof Mqtt5ConnAckException ||
+            throwable instanceof Mqtt5AuthException
+        ) {
+            return ctx.interruptMessagesWith(
+                new ExecutionFailure(500)
+                    .message("Endpoint connection failed")
+                    .key(FAILURE_ENDPOINT_CONNECTION_FAILED)
+                    .parameters(Map.of(FAILURE_PARAMETERS_EXCEPTION, throwable))
+            );
+        } else if (throwable instanceof Mqtt5DisconnectException) {
+            return ctx.interruptMessagesWith(
+                new ExecutionFailure(500)
+                    .message("Endpoint connection closed")
+                    .key(FAILURE_ENDPOINT_CONNECTION_CLOSED)
+                    .parameters(Map.of(FAILURE_PARAMETERS_EXCEPTION, throwable))
+            );
+        } else {
+            return ctx.interruptMessagesWith(
+                new ExecutionFailure(500)
+                    .message("Endpoint unknown error")
+                    .key(FAILURE_ENDPOINT_UNKNOWN_ERROR)
+                    .parameters(Map.of(FAILURE_PARAMETERS_EXCEPTION, throwable))
+            );
+        }
+    }
+
     private void closeClient(final Mqtt5RxClient client) {
-        client.disconnect().subscribe();
+        client.disconnect().onErrorComplete().subscribe();
```

Comment:
not even a log ? you're so cruel

Reply:
We don't log anything when the client connects so why doing it when it is disconnected?

Reply:
Actually errors here could be a normal behaviour. Connection is already disconnected for example. Don't think that will help to log those kind of error.